universal-application-tool-0.0.1/app/models/Application.java

```diff
@@ -33,9 +36,17 @@ public Application(Applicant applicant, Program program, LifecycleStage lifecycl
     this.applicant = applicant;
     setApplicantData(applicant.getApplicantData());
     this.program = program;
+    setVersion(program);
     this.lifecycleStage = lifecycleStage;
   }
 
+  private void setVersion(Program program) {
+    this.version =
+        program.getVersions().stream()
+            .max(Comparator.comparing(version -> version.id))
```

Comment:
Difficult, because we can go back in time - you might try `lastUpdatedTime` with `id` as a tiebreaker?

Reply:
hmm, I start feeling this is adding unnecessary complexity with unclear benefit. I will remove this and just go with always loading all questions.

Reply:
Removed.